examples/07_joining_tables_with_FuzzyJoin.py

```diff
@@ -0,0 +1,319 @@
+# so as to include only precise-enough matches:
+#
+X1 = fuzzy_join(
+    X,
+    gdppc,
+    left_on="Country",
+    right_on="Country Name",
+    match_score=0.35,
+    return_score=True,
+)
+print_worst_matches(X1, n=4)
+
+#################################################################
+# Matches that are not available (or precise enough) are marked as `NaN`.
+# We will remove them, as well as missing or unused information:
+
+mask = X1["GDP per capita (current US$)"].notna()
+X1 = X1[mask]
+y = np.ravel(y[mask])
+
+X1.drop(["matching_score"], axis=1, inplace=True)
+
+#################################################################
+#
+# We can finally plot and look at the link between GDP per capita
+# and happiness:
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+sns.set_context("notebook")
+
+plt.figure(figsize=(4, 3))
+ax = sns.regplot(x=X1["GDP per capita (current US$)"], y=y, logx=True)
+ax.set_ylabel("Happiness index")
+ax.set_title("Is having a higher GDP per capita sign of happiness?")
+plt.tight_layout()
+plt.show()
+
+#################################################
+# It seems that the happiest countries are those
+# having a high GDP per capita.
+# However, unhappy countries do not have only low levels
+# of GDP per capita. We have to search for other patterns.
+
+#################################################
+# 2. Joining life expectancy table
+# ................................
+#
+# Now let's include other information that may be relevant, such as
+# life expectancy table:
+X2 = fuzzy_join(
+    X1,
+    life_exp,
+    left_on="Country",
+    right_on="Country Name",
+    match_score=0.45,
+    how="left",
+)
+
+X2.head(3)
+
+#################################################
+# Let's plot this relation:
+plt.figure(figsize=(4, 3))
+fig = sns.regplot(
+    x=X2["Life expectancy at birth, total (years)"], y=y, fit_reg=True, ci=0
+)
+fig.set_ylabel("Happiness index")
+fig.set_title("Is having a higher life expectancy linked to happiness?")
+plt.tight_layout()
+plt.show()
+
+#################################################
+# It seems the answer is yes!
+# Countries with higher life expectancy are also happier.
+
+
+#################################################
+# 3. Joining legal rights strength table
+# ................................
+# .. topic:: Note:
+#
+#    Here, we use the `keep='left'` option to keep only the left key matching
+#    column, so as not to have unnecessary overlaping column with country names.
+#
+# And the table with a measure of legal rights strength in the country:
+X3 = fuzzy_join(
+    X2,
+    legal_rights,
+    left_on="Country",
+    right_on="Country Name",
+    match_score=0.45,
+    how="left",
+)
+
+X3.head(3)
+
+#################################################
+# Let's take a look at their correspondance in a figure:
+plt.figure(figsize=(4, 3))
+fig = sns.regplot(x=X3["Strength of legal rights index (0=weak to 12=strong)"], y=y)
```

Comment:
I tried using it, but the problem is that it will not work with our current minimal requirements. It requires the `statsmodels` package along with `seaborn` to work. So I used the other options available to get close to what the lowess would do..